deployment/ccip/changeset/cs_prerequisites.go

```diff
@@ -368,6 +374,59 @@ func deployPrerequisiteContracts(e deployment.Environment, ab deployment.Address
 	} else {
 		e.Logger.Infow("router already deployed", "chain", chain.String(), "addr", chainState.Router.Address)
 	}
+	if tokenPoolFactory == nil {
+		_, err := deployment.DeployContract(e.Logger, chain, ab,
+			func(chain deployment.Chain) deployment.ContractDeploy[*token_pool_factory.TokenPoolFactory] {
+				tpfAddr, tx2, contract, err2 := token_pool_factory.DeployTokenPoolFactory(
+					chain.DeployerKey,
+					chain.Client,
+					tokenAdminReg.Address(),
+					// There will always be at least one registry module deployed at this point.
+					// We just use the first one here. If a different RegistryModule is desired,
+					// users can run DeployTokenPoolFactoryChangeset with the desired address.
+					registryModules[0].Address(),
+					rmnProxy.Address(),
+					r.Address(),
+				)
+				return deployment.ContractDeploy[*token_pool_factory.TokenPoolFactory]{
+					Address: tpfAddr, Contract: contract, Tx: tx2, Tv: deployment.NewTypeAndVersion(TokenPoolFactory, deployment.Version1_5_1), Err: err2,
+				}
+			},
+		)
+		if err != nil {
+			e.Logger.Errorw("Failed to deploy token pool factory", "chain", chain.String(), "err", err)
+			return err
+		}
+	} else {
+		e.Logger.Infow("Token pool factory already deployed", "chain", chain.String(), "addr", tokenPoolFactory.Address)
+	}
+	// FactoryBurnMintERC20 is a contract that gets deployed by the TokenPoolFactory.
+	// We deploy it here so that we can verify it. All subsequent user deployments would then be verified.
+	if factoryBurnMintERC20 == nil {
```

Comment:
can you make it optional? I think we always might not need tokenPoolFactory and factoryBurnMintERC20 deployment. you can use [PrerequisiteOpt](https://github.com/smartcontractkit/chainlink/blob/develop/deployment/ccip/changeset/cs_prerequisites.go#L93)

Reply:
https://github.com/smartcontractkit/chainlink/pull/17179/commits/13a19e33559dcc28605a712eefaee832e9e1a453